addons/spreadsheet/static/tests/utils/pivot.js

```diff
@@ -8,13 +8,25 @@ import { getBasicServerData, getBasicPivotArch } from "./data";
 import { createModelWithDataSource } from "./model";
 import { waitForDataLoaded } from "@spreadsheet/helpers/model";
 import { helpers } from "@odoo/o-spreadsheet";
-const { parseDimension } = helpers;
+const { parseDimension, isDateField } = helpers;
 
 /**
  * @typedef {import("@spreadsheet").OdooSpreadsheetModel} OdooSpreadsheetModel
  * @typedef {import("@spreadsheet").Zone} Zone
  */
 
+function addEmptyGranularity(dimensions, fields) {
+    return dimensions.map((dimension) => {
+        if (dimension.name !== "id" && isDateField(fields[dimension.name])) {
```

Comment:
`id` cannot be inserted from the ui, reason why this helper is different from the one of `pivot_init_callback`